assembl/static2/js/app/pages/brightMirror.jsx

```diff
@@ -1,6 +1,68 @@
-// @flow
 import React from 'react';
+import { compose, graphql } from 'react-apollo';
+import Loader from '../components/common/loader';
+import ThematicsQuery from '../graphql/ThematicsQuery.graphql';
+import withLoadingIndicator from '../components/common/withLoadingIndicator';
+import { browserHistory } from '../router';
+import { get } from '../utils/routeMap';
 
-const BrightMirror = () => <div />;
+class BrightMirror extends React.Component {
+  componentDidMount() {
+    const { data, params } = this.props;
+    const isParentRoute = !params.themeId || false;
+    // After fetching thematics, redirect to the only BM thematic
+    if (isParentRoute && data.thematics.length > 0) {
+      const themeId = data.thematics[0].id;
+      browserHistory.push(`${get('idea', { slug: params.slug, phase: params.phase, themeId: themeId })}`);
+    }
+  }
 
-export default BrightMirror;
\ No newline at end of file
+  render() {
+    const { identifier, loading, children, params } = this.props;
+    const isParentRoute = !params.themeId || false;
+    if (isParentRoute) {
+      return (
+        <div className="debate">
+          {loading && isParentRoute && <Loader color="black" />}
+        </div>
+      );
+    }
+    const childrenElm = React.Children.map(children, child =>
+      React.cloneElement(child, {
+        id: params.themeId,
+        identifier: identifier
+      })
+    );
+    return (
+      <div>
+        <section className="debate-section">{childrenElm}</section>
+      </div>
+    );
+  }
+}
+
+export default compose(
+  graphql(ThematicsQuery, {
+    props: ({ data }) => {
+      if (data.loading) {
+        return {
+          loading: true
+        };
+      }
+
+      if (data.error) {
+        return {
+          hasErrors: true,
```

Comment:
maybe return the error aswell

Reply:
And do what with it? Don't return the error if you don't do anything with it. You should instead add a component that shows a user friendly message in case of error and depending of the case, allow the user to retry the query for example with a Retry button that do a refetch.